usr/src/cmd/dumpadm/svc-dumpadm

```diff
@@ -100,12 +101,17 @@ fi
 # how to modify the dump settings.
 #
 if [ -x /usr/sbin/dumpadm ]; then
-	/usr/sbin/dumpadm -u || $SMF_EXIT_ERR_CONFIG
+	/usr/sbin/dumpadm -u || exit $SMF_EXIT_ERR_CONFIG
 else
 	echo "WARNING: /usr/sbin/dumpadm is missing or not executable" >& 2
 	exit $SMF_EXIT_ERR_CONFIG
 fi
 
+if [[ -f /var/crash/volatile/keyfile ]]; then
+	/usr/sbin/dumpadm -k /var/crash/volatile/keyfile || \
```

Comment:
Use `$DUMPADM_SAVDIR` instead of `/var/cash/volatile`, but do this after line 116.